tools/checkpoint_training/gpt2_base_cn_poetry_training.ipynb

```diff
@@ -0,0 +1,340 @@
+{
+ "nbformat": 4,
+ "nbformat_minor": 0,
+ "metadata": {
+  "colab": {
+   "provenance": [],
+   "machine_shape": "hm"
+  },
+  "kernelspec": {
+   "name": "python3",
+   "display_name": "Python 3"
+  },
+  "language_info": {
+   "name": "python"
+  },
+  "accelerator": "GPU",
+  "gpuClass": "premium"
+ },
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Install KerasNLP and Import Dependencies."
+   ],
+   "metadata": {
+    "id": "AmJ2n25CtdiL"
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "fDG7RavCK0cq",
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "outputId": "a5d28081-e73c-4ca3-b0c2-8744bda75ce6"
+   },
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
+      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.8/5.8 MB\u001b[0m \u001b[31m49.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
+      "\u001b[?25h  Building wheel for keras-nlp (setup.py) ... \u001b[?25l\u001b[?25hdone\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip install -q -U git+https://github.com/keras-team/keras-nlp.git@master"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "import keras_nlp\n",
+    "import tensorflow as tf\n",
+    "from tensorflow import keras"
+   ],
+   "metadata": {
+    "id": "w-cqlGMjZcnr"
+   },
+   "execution_count": null,
+   "outputs": []
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Load `GPT2CausalLM` from KerasNLP.\n",
+    "\n",
+    "`GPT2CausalLM` is basically a GPT2 model, followed by multiplying the outputs by the embedding matrix."
+   ],
+   "metadata": {
+    "id": "cUNLW0dmfztt"
+   }
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "# To speed up, we use preprocessor of length 256 instead of full length 1024.\n",
+    "preprocessor = keras_nlp.models.GPT2CausalLMPreprocessor.from_preset(\n",
+    "    \"gpt2_base_en\",\n",
```

Comment:
How could will the result be for chinese language with the default GPT checkpoints, how much chinese language text is in the pre-training data?

Have we checked out if there's any other checkpoints available with a permissive license?

Reply:
Chinese generation is fine, but poem writing requires finetuning. Re the portion of Chinese text in the pretraining data, I don't have a concrete number. But since the source is from web crawled text, my guess is the amount of Chinese data is decent. 

I found a few checkpoints on HuggingFace, in the future we can welcome contributors to share their checkpoints (a few folks have expressed interest).

Reply:
Poked around in the paper, nothing on Chinese content, but there was this.

```
Performance on this task was surprising to us, since we deliberately removed non-English
webpages from WebText as a filtering step. In order to confirm this, we ran a byte-level language detector2 on WebText
which detected only 10MB of data in the French language
which is approximately 500x smaller than the monolingual
French corpus common in prior unsupervised machine translation research.
```

The overall dataset is 40GB, for reference, so less then a tenth of a percent of the pre-training content was French.

Reply:
Also probably worth noting that while the authors were surprised that english -> french translation worked at all, they note it is still far from state of the art from where this was published 4 years ago.